src/core/var/rsingleTag.js

```diff
@@ -1,5 +1,5 @@
 define( function() {
 
 	// Match a standalone tag
-	return ( /^<([\w-]+)\s*\/?>(?:<\/\1>|)$/ );
+	return ( /^<([a-zA-Z][^\s\/\0>]*)\s*\/?>(?:<\/\1>|)$/ );
```

Comment:
Some comments on this regular expression:
* `\s` is too broad—we should instead use [`\x20\t\r\n\f`](http://www.w3.org/TR/css3-selectors/#whitespace) (assuming https://github.com/whatwg/html/issues/242 is accepted)
* It shouldn't start accepting `:`, which is special in XHTML—such text should go through the [namespace-aware `innerHTML` path](https://github.com/jquery/jquery/pull/1988#discussion_r22442173)
* It would be smaller if it were case-insensitive rather than explicitly matching `A-Z`. Altogether, I got good results with `/^<([a-z][^:\/\0>\x20\t\r\n\f]*)[\x20\t\r\n\f]*\/?>(?:<\/\1>|)$/i`.